File: game/app/object/hard_switch.rb
class HardSwitch < Trigger
  def initialize(col, row, args)
    super(col, row, args, col * TILE_SIZE + 4, row * TILE_SIZE + 4, 2, 2, 'object/hard_switch', 2, 1, img_gap: Vector.new(-3, -3))
  end

  def activate(activator)
    return false unless activator.is_a?(Box)

    @img_index = 1
    super(activator)
  end
end

File: src/object/hard_switch.lua
HardSwitch = setmetatable({}, Trigger)
HardSwitch.__index = HardSwitch

function HardSwitch.new(id, col, row, args)
  local self = Trigger.new(id, col, row, args, 7, 7, 2, 2, "object/hard_switch" .. (args[2] or "1"), Vector.new(-5, -5), 2, 1)
  setmetatable(self, HardSwitch)
  self.serializable_attrs = {"img_index", "active"}
  return self
end

function HardSwitch:activate(activator)
  if getmetatable(activator) ~= Box then
    return false
  end

  self.img_index = 2
  return Trigger.activate(self, activator)
end
